### DISCUSSION TITLE
ESP32, TFT_eSPI, ILI9488, and FT6236 CTP Configuration Problem

### DISCUSSION THREAD

**Discussion by GeorgeIoak** [ImpulseAdventure/GUIslice#452] [Q&A]
So I have the above items all working in PlatformIO and I wanted to take the next step and use GUIslice.

I am having a very hard time just taking the ex46_ard_togglebtn.ino example and getting it to compile. I currently am getting error(s) like this:
```
.pio/libdeps/esp32dev/GUIslice/src/GUIslice_drv_tft_espi.cpp: In function 'bool gslc_DrvGetTxtSize(gslc_tsGui*, gslc_tsFont*, const char*, gslc_teTxtFlags, int16_t*, int16_t*, uint16_t*, uint16_t*)':
.pio/libdeps/esp32dev/GUIslice/src/GUIslice_drv_tft_espi.cpp:394:31: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
     m_disp.setFreeFont((const GFXfont *)pFont->pvFont);
                               ^
.pio/libdeps/esp32dev/GUIslice/src/GUIslice_drv_tft_espi.cpp:394:25: error: expected primary-expression before 'const'
     m_disp.setFreeFont((const GFXfont *)pFont->pvFont);
```
In platformio.ini I have
```
[env:esp32dev]
platform = espressif32
;platform = https://github.com/platformio/platform-espressif32.git  ; will force to use the 2.0 board package
board = esp32dev
framework = arduino
lib_deps = 
  TFT_eSPI
  Wire
  adafruit/Adafruit BusIO
  adafruit/Adafruit GFX Library  ; needed for GUIslice
  adafruit/Adafruit ILI9341
  adafruit/Adafruit FT6206 Library  ; needed for GUIslice
  impulseadventure/GUIslice
```
as well as the build flags that worked with my original proejct.

in `.pio\libdeps\esp32dev\GUIslice\src\GUIslice_config.h` I have uncommented `#include "../configs/esp-tftespi-default-ft6206.h"`

I don't think I need the Adafruit ILI9341 but I've tried with and without.

I tried uncommenting `#define EXTRA_FONTS` with no difference.

I've done plenty of searching but I can't seem to find much information for help on this, especially if you're using PlatforIO so I'm hoping someone has some idea what I'm doing wrong which I'm sure is a dumb mistake on my part.

**Comment by GeorgeIoak** [comment]
I thought I would add that following the error trail you come up with this:
![image](https://user-images.githubusercontent.com/1467145/157566676-0bbacdef-e1ad-4bbe-ae9a-afe49db65d1f.png)

But adding `#include <Adafruit_GFX.h>' by uncommenting `#define EXTRA_FONTS` should have pulled `GFXfont` definition in because it's defined in `https://github.com/adafruit/Adafruit-GFX-Library/blob/master/gfxfont.h` which is included in `Adafruit_GFX.h` so I wonder if this is an issue in PlatformIO?

**Comment by GeorgeIoak** [comment]
What I've found today is that if I remove the TFT_eSPI build flags then I can get the example code to compile. I've been playing with the build flags without any luck but maybe someone else can help out as everything says this should work and it must just be some linking issue somewhere that's preventing this to compile.

My complete platformio.ini file is
```
[platformio]
;src_dir = examples/arduino/ex46_ard_togglebtn
;lib_dir = .

[env]
;lib_ldf_mode = deep ;chain+ ;i also tried deep

[env:esp32dev]
platform = espressif32
;platform = https://github.com/platformio/platform-espressif32.git  ; will force to use the 2.0 board package
board = esp32dev
framework = arduino
lib_deps =
  SPI
  Wire
  adafruit/Adafruit BusIO
  adafruit/Adafruit GFX Library
;  adafruit/Adafruit ILI9341
  impulseadventure/GUIslice
  TFT_eSPI
  adafruit/Adafruit FT6206 Library  ; needed for GUIslice

build_flags =
  -DUSER_CONFIG_LOADED
  -DUSER_CONFIG_INC_FILE
  -DUSER_CONFIG_INC_FNAME=\"../configs/esp-tftespi-default-ft6206.h\"
  # Now TFT_eSPI Build Flags
  -DUSER_SETUP_LOADED=1
  -DBOARD_HAS_PSRAM=1
  -DILI9488_DRIVER=1                           ; Select ILI9488 driver
  -DTFT_WIDTH=320                              ; Set TFT size
  -DTFT_HEIGHT=480
  -DTFT_MISO=19                                ; Define SPI pins
  -DTFT_MOSI=23
  -DTFT_SCLK=18
  -DTFT_CS=5
  -DTFT_DC=14                                  ; Data/Comand pin
  -DTFT_RST=13                                 ; Reset pin
  -DTFT_BL=12                                  ; LED backlight control pin
  -DTFT_BACKLIGHT_ON=HIGH                      ; Level to turn on backlight
;  -DTOUCH_CS=21                                ; Not used but needed for compile
  -DLOAD_GLCD=1                                ; Load Fonts
  -DLOAD_FONT2=1          ; Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
  -DLOAD_FONT4=1          ; Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
  -DLOAD_FONT6=1          ; Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
  -DLOAD_FONT7=1          ; Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
  -DLOAD_FONT8=1          ; Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
  -DSMOOTH_FONT=1
  -DSPI_FREQUENCY=27000000                     ; Set SPI frequency
```

I created a new workspace and instead of importing the example I created a new one and then copied and pasted the code into main.cpp. I also added `#include <Arduino.h>` to main.cpp

**Comment by Pconti31** [comment]
@GeorgeIoak the original problem is with the compiler not finding the definition for GFXfont.  Not sure why but when you use tft_eSPI library it's complete in itself and has its own define for GFXfont. You do not need any Adafruit libraries. No Adafruit_gfx, Adafruit_BusIO, or Adafruit_ILI9341. I'm not sure about Adafruit_FT6206 which you might still need  Adafruit_BusIO.
Paul--

**Reply by GeorgeIoak** [reply]
Hi @Pconti31 thanks for looking. I've done so much experimenting I've gotten myself confused on what's needed or not!

I did find that if I comment out just this build flag `-DUSER_SETUP_LOADED=1` then it will compile. I think since all the other flags are in place this doesn't negatively impact anything but I'm not sure.

I thought I needed the Adafruit_FT6206 for the capacitive touch screen but maybe that's getting called somewhere else in the GUIslice config. I'm commenting over on https://github.com/ImpulseAdventure/GUIslice/issues/425 trying to get the CTP configured since on the ESP32 you can easily move the hardware I2C pins.

So if TFT_eSPI has it's own GFXfont define how would we work around this other than what I stumbled across?

**Reply by DonpK** [reply]
@GeorgeIoak @Pconti31 If you're looking for an alternative to the basic Arduino IDE and it's new Ver. 2, I can suggest looking at Programino, https://www.programino.com/  . It's much more flexible and configurable that the Arduino IDEs and so far, I've not had any problems using it with my GUISlice projects.

**Comment by GeorgeIoak** [comment]
So I've made it past the compile issues (assuming nothing is wrong with the way it's compiling). I'm trying to run the `examples/arduino/ex46_ard_togglebtn` example and as you can see it is not being displayed correctly. The touch does appear to work because the sliders do not initially show up after booting. The touch is also off but that's a separate issue than just getting things to display properly.

I did have the screen and touch working properly using just TFT_eSPI so something I've done is wrong when I added GUIslice into the mix.

![2022-03-10 15 56 08](https://user-images.githubusercontent.com/1467145/157775815-b62b34be-390d-4453-881d-5c17bf6a5f6f.jpg)

**Comment by Pconti31** [comment]
@GeorgeIoak I don't use PlatformIO so I would need to download it.   Before investigating PlatformIO further have you tried using Arduino's IDE with GUIslice.  Using Arduino IDE would pinpoint if its config of GUIslice or PlatformIO issue.  Config of GUIslice maybe I can help with; PlatformIO not so much...  You will need to wait on Calvin @ImpulseAdventure for that.

I should also note that Github doesn't send email notices for Discussions so posting in Issues section sends email so it will flag Calvin and at least he will know you have a problem.    
Paul--

**Reply by GeorgeIoak** [reply]
@Pconti31 TBH, I have kind of a love/hate relationship with PlatformIO. When it works it is so much better than working with the Arduino IDE but many times over the years it has bitten me. That's why I decided to post in the discussions first before opening an issue. I'm more of a hardware guy so I don't program regularly and can easily forget things!

I'm on Windows 10 and just opened that example project with Arduino 1.8.16 and it looks like the ESP32 goes into fault and ends up in an endless loop. The serial console shows this:
```
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1184
load:0x40078000,len:12804
ho 0 tail 12 room 4
load:0x40080400,len:3032
entry 0x400805e4
```

In `GUIslice_config.h` I have uncommented
`#include "../configs/esp-tftespi-default-ft6206.h"`

but I guess I need some coffee because I'm missing where I'm supposed to define the SPI and TFT connections. I see
 ` #define DRV_DISP_TFT_ESPI         // bodmer/TFT_eSPI`
in the config file which I guess hands off the pin configuring to TFT_eSPI

If that's correct then in `TFT_eSPI User_Setup_Select.h` I have a call to:
`#include <../TFT_eSPI_Setups/ILI9488_4SPI_AquaStat.h>`
which has all of my configuration and definitely works when I'm just using TFT_eSPI

Hopefully that helps explain the situation better. If you don't see anything I've missed then maybe I have run across a bug and need to open an issue?

**Comment by Pconti31** [comment]
@GeorgeIoak Well, I confess you are the edge of my knowledge here but I don't think User_Setup_Select.h is used by GUIslice.
I think it will pull in User_Setup.h instead.   This  would cause different pins to be defined.  Maybe why you are also having a problem with Platformio.
Try editing User_Setup .h to point to User_Setup_Select.h  and comment out everything else. Make sure User_Setup_Select.h doesn't point back to User_Setup.h!  If that works with Arduino IDE use that file also with Platformio.
Paul--

**Reply by GeorgeIoak** [reply]
@Pconti31 Well thanks for at least taking a look and bouncing ideas off of me. Before I tried to add GUIslice to the mix I followed this to configure TFT_eSPI, https://github.com/Bodmer/TFT_eSPI#tips, and that was/is working perfectly with my original Arduino environment sketch as well as the beginnings of a custom program in PlatformIO.

I'm not sure of the details on how GUIslice blends in TFT_eSPI but it obviously relies heavily on it for the underlying work when interfacing with the LCD.

You might be onto something in the issue being with the pin definitions so maybe I'll go back and try to not use the custom config file and see if GUIslice can follow along then.

This is looking more and more to be an actual bug but I really thought there were more people out there doing what I'm doing and could let me know if they see something wrong.

**Comment by Pconti31** [comment]
@GeorgeIoak I quick follow up.  A scan of TFT_eSPI shows that its main file brings in only User_Setup_Select.h which has the default of including User_Setup.h which I assume you commented out.  
A scan of GUIslice shows it doesn't reference either header file and simply relies upon the user placing one or the other in their project file. My Builder code only generates include for User_Setup.h which now in my mind is wrong.  Guess I'll do another update to Builder.
Doesn't explain your problem though.
Paul--

**Reply by GeorgeIoak** [reply]
Yes, I did comment out including User_Setup in User_Setup_Select.h so the only call was to my custom setup. Looking at TFT_eSPI.h I see that it calls USer_Setup_Select.h so this all should have worked.

I just ran another test and tried putting all the configuration into TFT_eSPI's User_Setup.h and now I get a different error in Arduino:
```
In file included from C:\Users\george\OneDrive\Documents\Arduino\libraries\GUIslice\src\GUIslice_drv_tft_espi.cpp:47:
C:\Users\george\OneDrive\Documents\Arduino\libraries\TFT_eSPI/TFT_eSPI.h:391:25: error: 'TFT_WIDTH' was not declared in this scope
   TFT_eSPI(int16_t _W = TFT_WIDTH, int16_t _H = TFT_HEIGHT);
```
There's more but you get the drift. `TFT_HEIGHT` is defined in ILI9488.h which gets pulled in from the first define in User_Setup.h
`#define ILI9488_DRIVER`

So it's looking like GUIslice and TFT_eSPI really aren't playing nicely with each other. @ImpulseAdventure might want to take a look and see if he can provide some helpful pointers or I can open an issue on this if we agree that it's looking like a true bug rather than user error.

**Comment by Pconti31** [comment]
@GeorgeIoak Yes, I think you should open a new issue, start with just the Arduino IDE.  I just tested with my esp32 and adafruit display ili9431 using TFTe_SPI, ArduinoIDE 1.8.13 and just editing User_Setup.h and it worked fine so I'm at a lost.  Also, I was wrong about my Builder, it outputs #include <TFTe_SPI.h> not any of the User_Setups .h files.
Paul--

**Reply by GeorgeIoak** [reply]
Hmm, bummer for me that it works for you. I wonder if it has something to do with the ILI9488 screen I'm using vs the ILI9431 that GUIslice seems to default to.

`#include <TFTe_SPI.h>` seems like the correct header to include to me.

**Reply by GeorgeIoak** [reply]
@Pconti31 I filed a bug so hopefully it will get looked at. Not to hijack my own thread but a quick question for you. With your builder can you configure custom touch button shapes? For instance create an up arrow or triangle and use it as a touch button?

**Reply by Pconti31** [reply]
Yes. Just pick image button. You can create images to load from C arrays or load BMP files from spiffs filesystem. 

C arrays can be created from BMP or png, or jpg, using my utility GUI_Image2C utility. See discussions 342 Image utilities for GUIslice for more details. The builder will read and parse the C file and correctly setup the image button for you.  if you need help post to my GUIslice Builder respository.
Paul--

**Reply by GeorgeIoak** [reply]
OK, that sounds perfect, now all I need is to get GUIslice working for me!

**Comment by ImpulseAdventure** [comment]
Hey @GeorgeIoak --

I just tried modifying the GUIslice default platformio.ini file to match the settings you posted in https://github.com/ImpulseAdventure/GUIslice/discussions/452#discussioncomment-2334910

I could get it to compile OK on my end (without the GFXfont issues) and without adding any GFX lib_deps.
The main differences I noted was that your config appeared to have added the Adafruit GFX libs but missing the following:
`-D LOAD_GFXFF`

I attached a modified platformio.ini file. Please try copying it over the platformio.ini file in your GUIslice root folder.
Then, within vscode, try using the PIO build under the new config "ESP32-TFT_eSPI-FT6206-loak".

Does that build any better for you?

thanks
[platformio.zip](https://github.com/ImpulseAdventure/GUIslice/files/8236853/platformio.zip)

**Reply by GeorgeIoak** [reply]
Hi @ImpulseAdventure 

Thanks for taking a stab at this. I have made some minor progress over in the Arduino IDE but I wanted to use PlatformIO as there's a bigger project I've started in that. 

I took your `platformio.ini` and overwrote my file. I then switched to the "environment" that you created and tried to compile and I'm getting an odd error
```
> Executing task in folder AquaStat: C:\Users\george\.platformio\penv\Scripts\platformio.exe run --environment ESP32-TFT_eSPI-FT6206-loak <

Processing ESP32-TFT_eSPI-FT6206-loak (platform: espressif32; board: featheresp32; framework: arduino)
----------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/featheresp32.html
PLATFORM: Espressif 32 (3.5.0) > Adafruit ESP32 Feather
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 3.10006.210326 (1.0.6)
 - tool-esptoolpy 1.30100.210531 (3.1.0)
 - toolchain-xtensa32 2.50200.97 (5.2.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ deep, Compatibility ~ soft
Found 42 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <TFT_eSPI> 2.4.42+sha.39b0d6a
|   |-- <SPIFFS> 1.0
|   |   |-- <FS> 1.0
|   |-- <FS> 1.0
|   |-- <SPI> 1.0
|-- <Wire> 1.0.1
|-- <Adafruit FT6206 Library> 1.0.6
|   |-- <Wire> 1.0.1
Error: Nothing to build. Please put your source code files to 'C:\Users\george\Box\Design\Villara\Thermostat\Design\Software\AquaStat\examples\arduino\ex04_ard_ctrls' folder
========================================================= [FAILED] Took 3.50 seconds =========================================================
Environment                 Status    Duration
--------------------------  --------  ------------
ESP32-TFT_eSPI-FT6206-loak  FAILED    00:00:03.498
=================================================== 1 failed, 0 succeeded in 00:00:03.498 =================================================== 
The terminal process "C:\Users\george\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 'ESP32-TFT_eSPI-FT6206-loak'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```
![image](https://user-images.githubusercontent.com/1467145/158001400-0441b8a8-12a7-48fd-962d-5709108a0d52.png)

I'm thinking this is the gods telling me that it's Friday night and I'd be better off on the couch drinking a beer so I'm not going to mess with fate and dig into this as it must be something dumb I've done (again). I'll definitely be back in the morning though to tackle this beast.

**Reply by ImpulseAdventure** [reply]
Hey George! I can appreciate the feeling trying to debug these things on a Friday night :)

It looks like you might be attempting to build this in your own project… if so, my recommendation is that we start by building the default GUIslice environment first, just to ensure we have a consistent startpoint.

If you haven’t already, can you try the following:
- clone / download a fresh copy of GUIslice from GitHub
- using vscode from within the fresh GUIslice library directory, try to build one of the PIO environments built-in to GUIslice. Do they compile?
- copy the `platformio.ini` file from my previous posting here over top of the default file from the repo
- within vscode, try to build the new env I created: `ESP32-TFT_eSPI-FT6206-loak`. Does it compile?

Thanks

**Reply by GeorgeIoak** [reply]
Good Morning @ImpulseAdventure (or at least it's morning for me)

Yes, last night I forgot that I had created a new workspace with just the GUIslice example but I didn't try your `platformio.ini` in that workspace, I guess I went for broke and used the project I was working on, sorry about that.

So I did as you suggested and cloned the repo and opened it in VS Code. Of course PIO through some false build errors but I've recompiled a bunch of times and am left with this:
```
Building in release mode
Compiling .pio\build\CustomConfig\src\ex04_ard_ctrls.ino.cpp.o
Linking .pio\build\CustomConfig\firmware.elf
Checking size .pio\build\CustomConfig\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [===       ]  28.8% (used 2359 bytes from 8192 bytes)
Flash: [==        ]  16.2% (used 42388 bytes from 261120 bytes)
============================================================================ [SUCCESS] Took 2.31 seconds ============================================================================

Environment                Status    Duration
-------------------------  --------  ------------
Arduino-ILI9341-STMPE610   SUCCESS   00:00:02.283
Arduino-ILI9341-FT6206     SUCCESS   00:00:02.310
Arduino-mcufriend-NoTouch  SUCCESS   00:00:02.154
Arduino-mcufriend-4wire    SUCCESS   00:00:02.273
Teensy-ILI9341_t3-NoTouch  FAILED    00:00:02.940
Teensy-ILI9341_t3-XPT2046  SUCCESS   00:00:02.894
ESP32-TFT_eSPI-STMPE610    SUCCESS   00:00:02.808
M5Stack-M5Stack-NoTouch    SUCCESS   00:00:03.235
CustomConfig               SUCCESS   00:00:02.308
======================================================================= 1 failed, 8 succeeded in 00:00:23.205 ======================================================================= 
The terminal process "C:\Users\george\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit code: 1.
```
Scrolling up I find the cause of that failure to be:
```
Compiling .pio\build\Teensy-ILI9341_t3-NoTouch\lib80a\SdFat\FatLib\FatFileSFN.cpp.o
In file included from D:/GitHub/GUIslice/examples/arduino/ex04_ard_ctrls/ex04_ard_ctrls.ino:22:0:
src/GUIslice.h:75:4: error: #error "CONFIG: DRV_TOUCH_NONE is not compatible with FEATURE_INPUT. Consider selecting DRV_TOUCH_INPUT in GUIslice config"
   #error "CONFIG: DRV_TOUCH_NONE is not compatible with FEATURE_INPUT. Consider selecting DRV_TOUCH_INPUT in GUIslice config"
    ^
Compiling .pio\build\Teensy-ILI9341_t3-NoTouch\lib80a\SdFat\FatLib\FatFormatter.cpp.o
Compiling .pio\build\Teensy-ILI9341_t3-NoTouch\lib80a\SdFat\FatLib\FatName.cpp.o
Compiling .pio\build\Teensy-ILI9341_t3-NoTouch\lib80a\SdFat\FatLib\FatPartition.cpp.o
Compiling .pio\build\Teensy-ILI9341_t3-NoTouch\lib80a\SdFat\FatLib\FatVolume.cpp.o
*** [.pio\build\Teensy-ILI9341_t3-NoTouch\src\ex04_ard_ctrls.ino.cpp.o] Error 1
```

Now, I don't really care about that build but I thought you should know that a fresh build doesn't successfully compile for all environments. Just a suggestion but I wonder if it might be better to not include all of the different environments in 1 platformio.ini file. I know at first when I looked at it I was a bit overwhelmed and it also causes some issues when you go to compile (or maybe I'm doing something wrong)
![image](https://user-images.githubusercontent.com/1467145/158028628-f81d99c5-f127-4208-aa31-325c7e35410c.png)

So, the good news is that after copying over your modified `platformio.ini` file I was able to successfully compile! I assume that since this is working from the just cloned GUIslice repo that all of the defaults for GUIslice and TFT_eSPI are in place and the only customization is inside the `platformio.ini` which is how things are supposed to work!

I did run into the endless reboot once I uploaded the example to my board but that's because my project is based on the ESP32 and doesn't have I2C on the default pins. We're discussing options/"work arounds" for this over in https://github.com/ImpulseAdventure/GUIslice/issues/425 but we need to be able to pass in the I2C pin definitions. I get that in the old days those were fixed but once Arduino started adding support for other processors libraries should have been updated to allow for a little more flexibility.

Now I need to go back to my project (that worked before GUIslice was added in) and see if I only have to add that one line you added (that was the only change, right?) or if there's more work to be done.

If you could take a look at Issue 425 I would appreciate it though since with the touch being off I can't really do any testing with GUIslice since I'm using the FT6236 CTP.

**Reply by GeorgeIoak** [reply]
@ImpulseAdventure I've been doing some experimenting with the PIO configurations and I'm getting a bit confused on how all the libraries are interlinked and correctly called.

In your general installation instructions for PlatformIO at https://github.com/ImpulseAdventure/GUIslice/wiki/Installing-GUIslice-on-Arduino#installation-platformio you say to add GUIslice as a library which obviously makes sense. But in your example `platformio.ini` file you don't list GUIslice in any of the `lib_deps` statements.

If I try to start some scratch with one of my example projects I get an error compiling if I don't list `impulseadventure/GUIslice` which makes sense to me so how does your `platformio.ini` correctly bring in your library?

While experimenting with the `platformio.ini` today I was able to reproduce what I reported in https://github.com/ImpulseAdventure/GUIslice/discussions/452#discussioncomment-2335499 above but I'm not able to figure what caused it so I have more work to do to try and pinout what changes cause that to happen.

**Comment by ImpulseAdventure** [comment]
FYI -- I have now posted an update to the Teensy config that should clean up the compilation error within that particular PIO environment (should not affect your build).

As for building the project within PlatformIO, I haven't run the flow for a while so I can't recall the specific steps off-hand to demonstrate how one integrates the INI config (I will need to followup separately as I believe it may have involved setting LDF mode). But you are correct: one would not be importing the platformio.ini file verbatim from GUIslice. The platformio file within the repo is intended to test a number of configurations within the default GUIslice library itself. It would be great if these IDEs enabled a simpler means of having a library (such as GUIslice) support a wide range of devices / displays, but unfortunately some extra steps are often required.

I am leaving on vacation tomorrow so unfortunately I don't know if I will get a chance to test and document the flow for you before I leave. In the meantime, I would recommend ensuring that your project works correctly within Arduino IDE (per the updates I've posted for you in issue #425 for example). After I return, I should be able to post the steps to get it to compile within PIO. Note that the majority of users and associated testing has been done in the Arduino IDE, though PIO has also been successfully used too.

thx
